src/ui/uiLayer.js

```diff
@@ -30,12 +30,15 @@ geo.gui.uiLayer = function (arg) {
    */
   ////////////////////////////////////////////////////////////////////////////
   this.createWidget = function (widgetName, arg) {
-
     var newWidget = geo.createWidget(
       widgetName, m_this, m_this.renderer(), arg);
 
-    m_this.addChild(newWidget);
-    newWidget._init();
+    // We only want top level widgets to be a child of the uiLayer
+    if (!('parent' in arg)) {
+      m_this.addChild(newWidget);
+    }
+
+    newWidget._init(arg); // @todo shouldn't this be maintained in the object?
```

Comment:
The `_init` and `_exit` methods are fairly inconsistently managed.  The convention has largely been that the parent object calls them for the child.  I would like to revisit this api once we get around to restructuring the classes for #418.